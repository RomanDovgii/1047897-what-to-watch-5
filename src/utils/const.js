export const MOVIES_COUNT = 9;
export const MAXIMUM_RAITING = 10;
export const MINIMUM_STARRING = 5;
export const MAXIMUM_STARRING = 10;
export const MAXIMUM_RAITING_COUNT = 1000;
export const MINIMUM_DESCRIPTION = 1;
export const MAXIMUM_DESCRIPTION = 10;

export const GENRES = [`All genres`, `Comedies`, `Crime`, `Documentary`, `Dramas`, `Horror`, `Kids & Family`, `Romance`, `Sci-Fi`, `Thrillers`];
export const MOVIES = [`Aviator`, `Bohemian Rhapsody`, `Dardjeeling Limited`, `Fantastic Beasts The Crimes of Grindelwald`, `Johnny English`, `Macbeth`, `Midnight Special`, `Mindhunter`, `Moonrise Kingdom`, `No Country For Old Men`, `Orlando`, `Pulp Fiction`, `Revenant`, `Seven Years In Tibet`, `Shutter Island`, `Snatch`, `War of the Worlds`, `We need to talk about Kevin`, `What We Do in the Shadows`];
export const VIDEO_URLS = [`https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`, `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`];
export const DIRECTORS = [`Steven Spielberg`, `Martin Scorsese`, `Alfred Hitchcock`, `Quentin Tarantino`, `Stanley Kubrick`, `Christopher Nolan`, `Tim Burton`, `Spike lee`];
export const ACTORS = [`Tom Hanks`, `Will Smith`, `Leonardo DiCaprio`, `Robert De Niro`, `Tom Cruise`, `Samuel L. Jackson`, `Robert Downey, Jr.`, `Scarlett Johansson`, `Jennifer Lawrence`, `Nickole Kidman`, `Cate Blanchett`, `Meryl Streep`, `Charlize Theron`, `Amy Adams`, `Angelina Jolie`];

export const FISH_TEXT = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc sodales convallis elit et lacinia. Quisque quis cursus lectus. Praesent elementum semper felis ut pretium. Ut eu mi mollis, luctus purus ac, aliquet ipsum. Maecenas libero nunc, faucibus a porttitor eget, laoreet hendrerit diam. Aliquam eget purus sit amet augue feugiat efficitur in et augue. Curabitur nec nulla a mauris maximus semper vel in dui. Vivamus mollis urna ex, id facilisis velit facilisis tincidunt. Maecenas vitae congue massa. Maecenas vel tortor justo.`;

export const GRAND_BUDAPEST_HOTEL = {
  name: `The Grand Budapest Hotel`,
  genre: `Drama`,
  release: 2014,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Tom Cruise`],
  rating: 8.9,
  ratingsCount: 240,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  isMyList: false
};

export const CatalogHeadingVariant = {
  CATALOG: `Catalog`, // скрыт
  MOVIE_PAGE: `More like this` // не скрыт
};
