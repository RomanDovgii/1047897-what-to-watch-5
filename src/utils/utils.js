import React from "react";
import {GENRES, MOVIES, VIDEO_URLS, DIRECTORS, ACTORS, MOVIES_COUNT, MAXIMUM_RATING, MAXIMUM_RATING_COUNT, FISH_TEXT, ActorsCount, Description, Paragraph, Duration, Id, CommentCount, MAXIMUM_SIMIALAR_MOVIES, MAXIMUM_DISPLAYED_ACTORS, RatingSystem, ALL_GENRE} from "./const";

const getRandomElementFromArray = (array) => {
  const max = array.length;
  const randomNumber = Math.floor(Math.random() * max);

  return array[randomNumber];
};

const getRandomIntegerNumber = (firstNumber, secondNumber = 0) => {
  const max = Math.max(firstNumber, secondNumber);
  const min = Math.min(firstNumber, secondNumber);

  return Math.floor(Math.random() * (max - min) + min);
};

const getRandomBoolean = () => {
  return (Math.random() >= 0.5);
};

const generateRating = () => {
  return Math.round((Math.random() * MAXIMUM_RATING) * 1e1) / 1e1;
};

const generateActors = () => {
  const actors = [];

  const starringCount = getRandomIntegerNumber(ActorsCount.MINIMUM, ActorsCount.MAXIMUM);

  for (let i = 0; i < starringCount; i++) {
    actors.push(getRandomElementFromArray(ACTORS));
  }

  return actors;
};

const generateParagraphText = () => {
  const randomSentences = [];
  const sentences = FISH_TEXT.split(`. `);
  const descriptionLength = getRandomIntegerNumber(Description.MINIMUM, Description.MAXIMUM);

  for (let i = 0; i < descriptionLength; i++) {
    randomSentences.push(getRandomElementFromArray(sentences));
  }

  const text = randomSentences.join(`. `);

  return text;
};

const generateDescriptionParagraphs = () => {
  const paragraphsCount = getRandomIntegerNumber(Paragraph.MINIMUM, Paragraph.MAXIMUM);
  let paragraphs = [];

  for (let i = 0; i < paragraphsCount; i++) {
    paragraphs.push(generateParagraphText());
  }

  return paragraphs;
};

const generateRelease = () => {
  const date = new Date();

  return date.getFullYear();
};

const generateMovie = () => {
  return {
    name: getRandomElementFromArray(MOVIES),
    genre: getRandomElementFromArray(GENRES),
    release: generateRelease(),
    descriptionParagraphs: generateDescriptionParagraphs(),
    director: getRandomElementFromArray(DIRECTORS),
    actors: generateActors(),
    rating: generateRating(),
    ratingsCount: getRandomIntegerNumber(MAXIMUM_RATING_COUNT),
    videoUrl: getRandomElementFromArray(VIDEO_URLS),
    duration: getRandomIntegerNumber(Duration.MINIMUM, Duration.MAXIMUM),
    id: getRandomIntegerNumber(Id.MINIMUM, Id.MAXIMUM),
    isMyList: getRandomBoolean()
  };
};

export const generateComment = () => {
  return {
    text: generateParagraphText(),
    rating: generateRating(),
    name: getRandomElementFromArray(ACTORS),
    date: new Date()
  };
};

const generateComments = (count) => {
  return new Array(count).fill().map(generateComment);
};

export const generateCommentsById = (movieIds) => {
  const comments = movieIds.reduce((accumulator, id) => {
    const number = getRandomIntegerNumber(CommentCount.MINIMUM, CommentCount.MAXIMUM);
    accumulator[id] = generateComments(number);
    return accumulator;
  }, {});

  return comments;
};

export const filterMoviesForMyList = (movies) => {
  return movies.filter((movie) => movie.isMyList);
};

export const generateMovies = () => {
  return new Array(MOVIES_COUNT).fill().map(generateMovie);
};

export const generateVideoType = (url) => {
  const urlPieces = url.split(`.`);
  const lastIndex = urlPieces.length - 1;

  return `video/${urlPieces[lastIndex]}`;
};

export const filterMoviesByGenre = (movies, genre) => {
  const filteredMovies = movies.slice().filter((movie) => movie.genre === genre);

  let finalArray = filteredMovies;

  if (filteredMovies.length < MAXIMUM_SIMIALAR_MOVIES) {
    const difference = MAXIMUM_SIMIALAR_MOVIES - filterMoviesByGenre.length;
    const additionalMovies = movies.slice().filter((movie) => movie.genre !== genre).slice(difference);
    finalArray = filteredMovies.concat(additionalMovies);
  }

  return finalArray;
};

export const generateDurationString = (duration) => {
  return `${Math.floor(duration / 60)}h ${duration % 60}m`;
};

export const generateActorsText = (actors) => {
  const actorsMaxIndex = actors.length - 1;

  return (
    <span className="movie-card__details-value">
      {actors.map((actor, i) => (
        <React.Fragment key={i}>
          {actor}{i < actorsMaxIndex ? `, ` : ``} {i < actorsMaxIndex ? <br/> : ``}
        </React.Fragment>
      ))}
    </span>);
};

export const generateStarringString = (actors) => {
  let text = ``;
  let end = ``;
  let actorsLocal = actors.slice();

  if (actorsLocal.length > MAXIMUM_DISPLAYED_ACTORS) {
    actorsLocal = actors.slice(0, MAXIMUM_DISPLAYED_ACTORS);

    end = ` and other`;
  }

  text = actorsLocal.join(`, `);

  text = text.slice(0, -2);

  text += end;

  return text;
};

export const generateRatingText = (rating) => {
  switch (true) {
    case rating <= RatingSystem.BAD:
      return `Bad`;
    case rating <= RatingSystem.NORMAL:
      return `Normal`;
    case rating <= RatingSystem.GOOD:
      return `Good`;
    case rating < RatingSystem.AWESOME:
      return `Very good`;
    case rating === RatingSystem.AWESOME:
      return `Awesome`;
    default:
      return `Rating is incorrect`;
  }
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const createGenresList = (movies) => {
  const genresNotSorted = [];

  movies.map((movie) => {
    genresNotSorted.push(movie.genre);
  });

  const genres = [];

  genresNotSorted.map((genre) => {
    if (!genres.includes(genre)) {
      genres.push(genre);
    }
  });

  return [ALL_GENRE, ...genres];
};
