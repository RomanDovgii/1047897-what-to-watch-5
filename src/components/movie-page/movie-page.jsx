import React from "react";
import {connect} from "react-redux";
import MovieCard from "./modules/movie-card";
import Catalog from "../catalog/catalog";
import Footer from "../footer/footer";
import {CatalogHeadingVariant, CatalogCallSource, MoreLikeThis} from "../../utils/const";
import {moviePageType} from "../types/types";
import withActiveTabs from "../hoc/with-active-tabs/with-active-tabs";
import {filterMoviesByGenre} from "../../store/selectors/genre-selector";
import {fetchSelectedMovie, fetchSelectedMovieComments} from "../../store/actions/api-actions";

const MovieCardWrapper = withActiveTabs(MovieCard);

const MoviePage = (props) => {
  const {onUserIconClick, onWTWLogoClick, onPlayButtonClick, selectedMovie, movies, comments, fetchMovie, isAuth} = props;

  const id = window.location.pathname.slice(7);

  return (
    <React.Fragment>
      <MovieCardWrapper
        onUserIconClick = {onUserIconClick}
        onWTWLogoClick = {onWTWLogoClick}
        onPlayButtonClick = {onPlayButtonClick}
        selectedMovie={selectedMovie}
        comments = {comments}
        id = {id}
        fetchMovie = {fetchMovie}
        isAuth = {isAuth}
      />

      <div className="page-content">
        <Catalog
          heading = {CatalogHeadingVariant.MOVIE_PAGE}
          movies = {movies.slice(MoreLikeThis.FIRST_INDEX, MoreLikeThis.LAST_INDEX)}
          source = {CatalogCallSource.MOVIE_PAGE}
        />
        <Footer/>
      </div>
    </React.Fragment>
  );
};

MoviePage.propTypes = moviePageType;

const mapStateToProps = ({DATA, USER}) => ({
  selectedMovie: DATA.selectedMovie,
  movies: filterMoviesByGenre(DATA),
  comments: DATA.comments,
  isAuth: USER.authorizationStatus
});

const mapDispatchToProps = (dispatch) => ({
  fetchMovie(id) {
    dispatch(fetchSelectedMovie(id));
    dispatch(fetchSelectedMovieComments(id));
  }
});

export {MoviePage};
export default connect(mapStateToProps, mapDispatchToProps)(MoviePage);
