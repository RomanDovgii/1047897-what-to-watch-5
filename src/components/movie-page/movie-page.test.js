import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import {noop} from "../../utils/utils";
import {MoviePage} from "./movie-page";

jest.requireActual(`react-router-dom`);
jest.mock(`react-router-dom`, () => ({
  useParams: jest.fn().mockReturnValue({id: 1}),
}));

const testComment = {
  id: 1,
  user: {
    id: 4,
    name: `Kate Muir`
  },
  rating: 8.9,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`
};
const testComments = [
  testComment,
  testComment,
  testComment,
  testComment
];
const testMovie = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  previewImage: `img/the-grand-budapest-hotel.jpg`,
  backgroundImage: `img/the-grand-budapest-hotel-bg.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://some-link`,
  previewVideoLink: `https://some-link`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  runTime: 99,
  genre: `Comedy`,
  released: 2014,
  isFavorite: false
};

const testMovies = [
  testMovie,
  testMovie,
  testMovie,
  testMovie,
  testMovie
];

it(
    `Should render correct MoviePage with authorized user`,
    () => {
      const renderer = new ShallowRenderer();
      const tree = renderer
      .render(
          <MoviePage
            onUserIconClick={noop}
            onWTWLogoClick={noop}
            onPlayButtonCLick={noop}
            fetchMovie={noop}
            onMyListClick={noop}
            selectedMovie={testMovie}
            movies={testMovies}
            comments={testComments}
            isAuth={true}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);

it(
    `Should render correct MoviePage with non-authorized user`,
    () => {
      const renderer = new ShallowRenderer();
      const tree = renderer
      .render(
          <MoviePage
            onUserIconClick={noop}
            onWTWLogoClick={noop}
            onPlayButtonCLick={noop}
            fetchMovie={noop}
            onMyListClick={noop}
            selectedMovie={testMovie}
            movies={testMovies}
            comments={testComments}
            isAuth={false}
          />
      );

      expect(tree).toMatchSnapshot();
    }
);
