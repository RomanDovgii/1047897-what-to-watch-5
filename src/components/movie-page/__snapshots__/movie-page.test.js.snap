// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correct MoviePage with authorized user 1`] = `
<React.Fragment>
  <WithLoading
    comments={
      Array [
        Object {
          "comment": "Lorem ipsum dolor sit amet",
          "date": "2019-05-08T14:13:56.569Z",
          "id": 1,
          "rating": 8.9,
          "user": Object {
            "id": 4,
            "name": "Kate Muir",
          },
        },
        Object {
          "comment": "Lorem ipsum dolor sit amet",
          "date": "2019-05-08T14:13:56.569Z",
          "id": 1,
          "rating": 8.9,
          "user": Object {
            "id": 4,
            "name": "Kate Muir",
          },
        },
        Object {
          "comment": "Lorem ipsum dolor sit amet",
          "date": "2019-05-08T14:13:56.569Z",
          "id": 1,
          "rating": 8.9,
          "user": Object {
            "id": 4,
            "name": "Kate Muir",
          },
        },
        Object {
          "comment": "Lorem ipsum dolor sit amet",
          "date": "2019-05-08T14:13:56.569Z",
          "id": 1,
          "rating": 8.9,
          "user": Object {
            "id": 4,
            "name": "Kate Muir",
          },
        },
      ]
    }
    fetchMovie={[Function]}
    id={1}
    isAuth={true}
    onMyListClick={[Function]}
    onUserIconClick={[Function]}
    onWTWLogoClick={[Function]}
    selectedMovie={
      Object {
        "backgroundColor": "#ffffff",
        "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
        "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
        "director": "Wes Andreson",
        "genre": "Comedy",
        "id": 1,
        "isFavorite": false,
        "name": "The Grand Budapest Hotel",
        "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
        "previewImage": "img/the-grand-budapest-hotel.jpg",
        "previewVideoLink": "https://some-link",
        "rating": 8.9,
        "released": 2014,
        "runTime": 99,
        "scoresCount": 240,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
          "Saoirse Ronan",
        ],
        "videoLink": "https://some-link",
      }
    }
  />
  <div
    className="page-content"
  >
    <Memo(Connect(Catalog))
      heading="More like this"
      movies={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          },
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          },
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          },
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          },
        ]
      }
      source="MOVIE_PAGE"
    />
    <Memo(Connect(Footer)) />
  </div>
</React.Fragment>
`;

exports[`Should render correct MoviePage with non-authorized user 1`] = `
<React.Fragment>
  <WithLoading
    comments={
      Array [
        Object {
          "comment": "Lorem ipsum dolor sit amet",
          "date": "2019-05-08T14:13:56.569Z",
          "id": 1,
          "rating": 8.9,
          "user": Object {
            "id": 4,
            "name": "Kate Muir",
          },
        },
        Object {
          "comment": "Lorem ipsum dolor sit amet",
          "date": "2019-05-08T14:13:56.569Z",
          "id": 1,
          "rating": 8.9,
          "user": Object {
            "id": 4,
            "name": "Kate Muir",
          },
        },
        Object {
          "comment": "Lorem ipsum dolor sit amet",
          "date": "2019-05-08T14:13:56.569Z",
          "id": 1,
          "rating": 8.9,
          "user": Object {
            "id": 4,
            "name": "Kate Muir",
          },
        },
        Object {
          "comment": "Lorem ipsum dolor sit amet",
          "date": "2019-05-08T14:13:56.569Z",
          "id": 1,
          "rating": 8.9,
          "user": Object {
            "id": 4,
            "name": "Kate Muir",
          },
        },
      ]
    }
    fetchMovie={[Function]}
    id={1}
    isAuth={false}
    onMyListClick={[Function]}
    onUserIconClick={[Function]}
    onWTWLogoClick={[Function]}
    selectedMovie={
      Object {
        "backgroundColor": "#ffffff",
        "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
        "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
        "director": "Wes Andreson",
        "genre": "Comedy",
        "id": 1,
        "isFavorite": false,
        "name": "The Grand Budapest Hotel",
        "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
        "previewImage": "img/the-grand-budapest-hotel.jpg",
        "previewVideoLink": "https://some-link",
        "rating": 8.9,
        "released": 2014,
        "runTime": 99,
        "scoresCount": 240,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
          "Saoirse Ronan",
        ],
        "videoLink": "https://some-link",
      }
    }
  />
  <div
    className="page-content"
  >
    <Memo(Connect(Catalog))
      heading="More like this"
      movies={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          },
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          },
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          },
          Object {
            "backgroundColor": "#ffffff",
            "backgroundImage": "img/the-grand-budapest-hotel-bg.jpg",
            "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.",
            "director": "Wes Andreson",
            "genre": "Comedy",
            "id": 1,
            "isFavorite": false,
            "name": "The Grand Budapest Hotel",
            "posterImage": "img/the-grand-budapest-hotel-poster.jpg",
            "previewImage": "img/the-grand-budapest-hotel.jpg",
            "previewVideoLink": "https://some-link",
            "rating": 8.9,
            "released": 2014,
            "runTime": 99,
            "scoresCount": 240,
            "starring": Array [
              "Bill Murray",
              "Edward Norton",
              "Jude Law",
              "Willem Dafoe",
              "Saoirse Ronan",
            ],
            "videoLink": "https://some-link",
          },
        ]
      }
      source="MOVIE_PAGE"
    />
    <Memo(Connect(Footer)) />
  </div>
</React.Fragment>
`;
